linuxbrew-common.sh

```diff
@@ -22,30 +22,52 @@ fi
 
 declare -i -r ABS_PATH_LIMIT=85
 
-get_brew_link() {
-  local brew_dirname="linuxbrew-$(date +%Y%m%dT%H%M%S)"
-  local brew_link="$(realpath .)/$brew_dirname"
-  local len=${#brew_link}
+log() {
+  echo >&2 "[$( date +%Y-%m-%dT%H:%M:%S )] $*"
+}
+
+fatal() {
+  log "$@"
+  exit 1
+}
+
+set_brew_timestamp() {
+  if [[ -z ${YB_BREW_TIMESTAMP:-} ]]; then
+    export YB_BREW_TIMESTAMP=$(date +%Y%m%dT%H%M%S)
+  fi
+}
+
+get_brew_path_prefix() {
+  set_brew_timestamp
+  local brew_dirname="$YB_BREW_DIR_PREFIX-$YB_BREW_TIMESTAMP"
+  local brew_path_prefix="$(realpath .)/$brew_dirname"
+  if [[ -n ${YB_BREW_SUFFIX:-} ]]; then
+    brew_path_prefix+="-$YB_BREW_SUFFIX"
+  fi
+  local len=${#brew_path_prefix}
   if [[ $len -gt $ABS_PATH_LIMIT ]]; then
-    echo "Linuxbrew link absolute path should be no more than $ABS_PATH_LIMIT bytes, but actual" \
-         "length is $len bytes: $brew_link" >&2
+    echo "Brew link absolute path should be no more than $ABS_PATH_LIMIT bytes, but actual" \
```

Comment:
Looks like it is not actually a link anymore?

Reply:
Done.